dags/mlflow-dag.py

```diff
@@ -0,0 +1,206 @@
+from airflow.decorators import task, dag
+from airflow.providers.google.cloud.hooks.bigquery import BigQueryHook
+
+from datetime import datetime
+import logging
+from mlflow.tracking.fluent import log_metric
+
+import pandas as pd
+import numpy as np
+from sklearn.model_selection import RepeatedStratifiedKFold
+from sklearn.model_selection import cross_val_score
+from lightgbm import LGBMClassifier
+
+import mlflow
+
+
+docs = """
+By default, Airflow stores all return values in XCom. However, this can introduce complexity, as users then have to consider the size of data they are returning. Futhermore, since XComs are stored in the Airflow database by default, intermediary data is not easily accessible by external systems.
+By using an external XCom backend, users can easily push and pull all intermediary data generated in their DAG in GCS.
+"""
+
+mlflow.set_tracking_uri('http://host.docker.internal:5000')
+try:
+    # Creating an experiment 
+    mlflow.create_experiment('census_prediction')
+except:
+    pass
+# Setting the environment with the created experiment
+mlflow.set_experiment('census_prediction')
+
+
+
+@dag(
+    start_date=datetime(2021, 1, 1),
+    schedule_interval=None,
+    catchup=False,
+    doc_md=docs
+)
+def using_gcs_for_xcom_ds():
+
+    @task
```

Comment:
It would be amazing to see this is a SQL decorator. Not sure if there's any thought on saving query results to pandas dataframes, but that seems like it would be a pretty obvious use case for ML, as you're doing here. Just being able to write this as:
```
@sql(to_pandas=True)
def load_data():
    """
    SELECT * FROM `bigquery-public-data.ml_datasets.census_adult_income`
    """
```

Reply:
Good call out, this occurred to me as well, but did not check to see if the decorator works with BigQuery yet.

Reply:
Yeah great idea